core/controllers/base_test.py

```diff
@@ -712,6 +794,43 @@ def test_controller_class_names(self):
         self.assertGreater(num_handlers_checked, 150)
 
 
+class IframeRestrictionTests(test_utils.GenericTestBase):
+
+    class MockHandler(base.BaseHandler):
+        def get(self):
+            iframe_restriction = self.request.get(
+                'iframe_restriction', default_value=None)
+            self.render_template(
+                'pages/about/about.html',
+                iframe_restriction=iframe_restriction)
+
+    def setUp(self):
+        super(IframeRestrictionTests, self).setUp()
+        self.signup(self.OWNER_EMAIL, self.OWNER_USERNAME)
+        self.owner_id = self.get_user_id_from_email(self.OWNER_EMAIL)
+        # Modify the testapp to use the mock handler.
+        self.testapp = webtest.TestApp(webapp2.WSGIApplication(
+            [webapp2.Route('/mock', self.MockHandler, name='MockHandler')],
+            debug=feconf.DEBUG,
+        ))
+
+    def test_responses_with_valid_iframe_restriction(self):
+        self.login(self.OWNER_EMAIL)
+        self.get_html_response('/mock')
+        self.get_html_response('/mock', params={'iframe_restriction': 'DENY'})
+        self.get_html_response(
```

Comment:
In these cases you should be checking for something in the HTML response that matches the desired iframe restriction, right?

Reply:
Done